components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSPortalAppMigrationTool.java

```diff
@@ -0,0 +1,376 @@
+
+    }
+
+    /**
+     * Migrate the different types of artifacts such as gadgets,widgets and layouts into newer version
+     */
+    private void migrateArtifactsInStore() {
+        File store = new File(tempDir);
+        File[] tenantStores = store.listFiles();
+        for (int i = 0; i < tenantStores.length; i++) {
+            if (tenantStores[i].isDirectory()) {
+                File[] artifactTypes = tenantStores[i].listFiles();
+                setTenantDomain(tenantStores[i].getName());
+                for (int artifactCount = 0; artifactCount < artifactTypes.length; artifactCount++) {
+                    if (artifactTypes[artifactCount].getName().equalsIgnoreCase(Constants.GADGET)
+                            || artifactTypes[artifactCount].getName().equalsIgnoreCase(Constants.WIDGET)) {
+                        new DSPortalAppMigrationTool().gadgetJSONUpdater(artifactTypes[artifactCount]);
+                    } else if (artifactTypes[artifactCount].getName().equalsIgnoreCase(Constants.LAYOUT)) {
+                        migrateLayoutsInStore(artifactTypes[artifactCount]);
+                    }
+                }
+                setTenantDomain(null);
+            }
+        }
+    }
+
+    /**
+     * Migrate layouts into newer version
+     *
+     * @param layouts file path of the layouts directory
+     */
+    private void migrateLayoutsInStore(File layouts) {
+        File[] listOflayouts = layouts.listFiles();
+        JSONParser parser = new JSONParser();
+        for (int layoutCount = 0; layoutCount < listOflayouts.length; layoutCount++) {
+            try {
+                JSONObject layoutObj = (JSONObject) parser.parse(new FileReader(
+                        listOflayouts[layoutCount].getAbsolutePath() + File.separator + Constants.INDEX_JSON));
+                updateDashboardBlocks(((JSONArray) layoutObj.get(Constants.BLOCKS)));
+                FileWriter file = new FileWriter(
+                        listOflayouts[layoutCount].getAbsolutePath() + File.separator + Constants.INDEX_JSON);
+                file.write(layoutObj.toJSONString());
+                file.flush();
+                file.close();
+            } catch (IOException e) {
+                log.error("Error in opening the file " + listOflayouts[layoutCount].getName(), e);
+            } catch (ParseException e) {
+                log.error("Error in parsing the index.json file in " + listOflayouts[layoutCount].getAbsolutePath());
+            }
+        }
+    }
+
+    /**
+     * Login to the portal of given url and return the session id
+     *
+     * @param url      url of the server
+     * @param username username for login to the portal
+     * @param password password for login to the portal
+     * @return String session id
+     */
+    private String loginToPortal(String url, String username, String password) {
+        String response = invokeRestAPI(url + Constants.USERNAME_QUERY + username + Constants.PASSWORD_QUREY + password,
```

Comment:
Throw the exception to the caller bcz at this stage anyhow the application cannot continue to operation.